File: projects/hello-world/helloworld.rb
#!/usr/bin/ruby
#+-----------------------------------------------------------------------------+
#| A contrived hello world program for the ruby programming language.          |
#| Written by Dmitry Makhnin.                                                  |
#+-----------------------------------------------------------------------------+

def main(argv)
    if argv.size == 0 then
        puts "It appears you have no arguments."
        puts "I'll say hello, world anyway!"
        return 1
    end

    argv.each { |arg|
        print arg + ' '
    }

    print "\n"

    return 0
end

exit main ARGV

File: projects/hello-world/helloworld.lua
--[[ 
     A hello world program written in Lua programming language.
     Written by Dmitry Makhnin.
--]]
function main(arg)
    if #arg < 1 then
        print "you wrote no arguments.";
        print "But hello anyway!";
        return 1;
    end

    for i = 1, #arg do
        io.write(arg[i], " ");
    end

    return 0;
end


os.exit(main(arg));
